### PR TITLE
BACKLOG-10418 update Text component to support number types

### PR DESCRIPTION
**Pull request by Taybou**

### CODE REVIEW

File: src/javascript/EditPanelContainer/EditPanel/EditPanel.utils.js [hunk 1]
```diff
@@ -15,11 +15,14 @@ export function getPropertiesToMutate(nodeData = {}, formValues = {}, fields = [
...
+            const value = formValues[key];
+            if (value) {
+                const fieldType = field.jcrDefinition.requiredType;
+
                 propsToSave.push({
                     name: key,
-                    type: field.jcrDefinition.requiredType,
-                    value: formValues[key],
+                    type: fieldType,
+                    value: fieldType === 'DECIMAL' || fieldType === 'DOUBLE' ? value && value.replace(',', '.') : value,
```
Comment (FBerthelot) [inline src/javascript/EditPanelContainer/EditPanel/EditPanel.utils.js]: is it i18n proof ?

File: src/javascript/EditPanelContainer/EditPanel/EditPanelContent/FormBuilder/SelectorTypes/Text/Text.jsx [hunk 2]
```diff
@@ -6,16 +6,27 @@ import {Input} from '../../../../../../DesignSystem/Input';
...
-        let {field, id} = this.props;
+        let {field, id, editorContext} = this.props;
         let {values, handleChange, handleBlur} = this.props.formik;
 
+        const fieldType = field.jcrDefinition.requiredType;
+        const isNumber = fieldType === 'DOUBLE' ||
+            fieldType === 'LONG' ||
+            fieldType === 'DECIMAL';
+
+        const decimalSeparator = editorContext.lang === 'en' ? ',' : '.';
```
Comment (FBerthelot) [inline src/javascript/EditPanelContainer/EditPanel/EditPanelContent/FormBuilder/SelectorTypes/Text/Text.jsx]: des tests sur ce genre de chose ?

File: src/javascript/EditPanelContainer/EditPanel/EditPanel.utils.js (hunk 1 above)
Comment (Taybou) [reply src/javascript/EditPanelContainer/EditPanel/EditPanel.utils.js]: No.

File: src/javascript/EditPanelContainer/EditPanel/EditPanelContent/FormBuilder/SelectorTypes/Text/Text.jsx (hunk 2 above)
Comment (Taybou) [reply src/javascript/EditPanelContainer/EditPanel/EditPanelContent/FormBuilder/SelectorTypes/Text/Text.jsx]: Yes, why not, I think it's more persistent from end-to-end, but no problem I'll add some tests.

**Pull request merged**